File: src/rules/cookies.js
const session = require('express-session');

function setCookies(res) {
  // ruleid: clipeus-insecure-cookie-httponly-false
  res.cookie('sid', token, { httpOnly: false });

  // ruleid: clipeus-insecure-cookie-secure-false
  res.cookie('sid', token, { secure: false, httpOnly: true });

  // ok: clipeus-insecure-cookie-httponly-false
  // ok: clipeus-insecure-cookie-secure-false
  res.cookie('sid', token, { httpOnly: true, secure: true, sameSite: 'strict' });
}

// ruleid: clipeus-insecure-cookie-secure-false
app.use(session({ secret: s, cookie: { secure: false } }));

// ok: clipeus-insecure-cookie-httponly-false
app.use(session({ secret: s, cookie: { httpOnly: true, secure: true } }));

File: src/rules/cookies.py
from flask import Flask

app = Flask(__name__)


def set_cookies(resp):
    # ruleid: patronus-insecure-cookie-flask
    resp.set_cookie("sid", token, httponly=False)

    # ruleid: patronus-insecure-cookie-flask
    resp.set_cookie("sid", token, secure=False)

    # ok: patronus-insecure-cookie-flask
    resp.set_cookie("sid", token, httponly=True, secure=True, samesite="Strict")


# ruleid: patronus-insecure-cookie-flask
app.config['SESSION_COOKIE_HTTPONLY'] = False

# ruleid: patronus-insecure-cookie-flask
app.config['SESSION_COOKIE_SECURE'] = False

# ok: patronus-insecure-cookie-flask
app.config['SESSION_COOKIE_HTTPONLY'] = True
